src/main/java/uk/gov/cshr/controller/IdentityController.java

```diff
@@ -3,16 +3,119 @@
 import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RestController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.cshr.domain.Identity;
+import uk.gov.cshr.domain.Role;
 import uk.gov.cshr.dto.IdentityDTO;
+import uk.gov.cshr.service.AuthenticationDetails;
 import uk.gov.cshr.service.security.IdentityDetails;
+import uk.gov.cshr.repository.IdentityRepository;
+import uk.gov.cshr.service.IdentityService;
+import uk.gov.cshr.service.RoleService;
 
-@RestController
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Controller
+@RequestMapping("/management/")
 public class IdentityController {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(RoleController.class);
+
+    @Autowired
+    private IdentityService identityService;
+
+    @Autowired
+    private RoleService roleService;
+
+    @Autowired
+    private IdentityRepository identityRepository;
+
+    @Autowired
+    private AuthenticationDetails authenticationDetails;
+
     @RequestMapping(value = "/identity", method = RequestMethod.GET)
     public IdentityDTO getIdentityDetailsfromAccessToken(Authentication authentication) {
         IdentityDetails identityDetails = (IdentityDetails) authentication.getPrincipal();
         return new IdentityDTO(identityDetails.getIdentity());
     }
+
+    @GetMapping("/identities")
+    public String identities(Model model) {
+        LOGGER.info("Listing all identities");
+
+        Iterable<Identity> identities = identityService.findAll();
+        model.addAttribute("identities", identities);
+        return "identityList";
+    }
+
+    @GetMapping("/identities/update/{uid}")
+    public String identityEdit(Model model,
+                           @PathVariable("uid") String uid) {
+
+        LOGGER.info("{} editing identity for uid {}", authenticationDetails.getCurrentUsername(),uid);
+
+        Optional<Identity> optionalIdentity = identityService.getIdentity(uid);
+        Iterable<Role> roles = roleService.findAll();
+
+        if (optionalIdentity.isPresent()){
+            Identity identity = optionalIdentity.get();
+            model.addAttribute("identity", identity);
+            model.addAttribute("roles", roles);
+            return "updateIdentity";
+        }
+
+        LOGGER.info("No identity found for uid {}", authenticationDetails.getCurrentUsername(), uid);
+        return "redirect:/management/identities";
+    }
+
+    @PostMapping("/identities/update")
+    public String identityUpdate(@RequestParam(value = "active", required = false) Boolean active, @RequestParam("roleId") ArrayList<String> roleId, @RequestParam("uid") String uid) {
+
```

Comment:
On the 'Update Identity' screen, if the user doesn't select any roles and clicks update, they will receive a 400 error:

```
There was an unexpected error (type=Bad Request, status=400).
Required ArrayList parameter 'roleId' is not present
``` 

We could catch this exception, maybe by defining something like this in the controller:
```
@ExceptionHandler(MissingServletRequestParameterException.class)
    public String handleMissingParams(MissingServletRequestParameterException ex) {
        LOGGER.ERROR(ex.getParameterName() + " parameter is missing");
        return "redirect:/management/identities";
    }
```

Other solutions possible, so maybe lets have a talk

Reply:
I've just made it optional  and checked for it being set within the function loop